jaeger-context/src/main/java/com/uber/jaeger/context/TracingUtils.java

```diff
@@ -14,10 +14,19 @@
 
 package com.uber.jaeger.context;
 
+import io.opentracing.util.GlobalTracer;
+
 import java.util.concurrent.ExecutorService;
 
 public class TracingUtils {
-  private static final TraceContext traceContext = new ThreadLocalTraceContext();
+  private static final TraceContext traceContext;
+
+  static {
+    if (!GlobalTracer.isRegistered()) {
+      throw new IllegalStateException("Please register a io.opentracing.util.GlobalTracer");
+    }
+    traceContext = new ActiveSpanSourceTraceContext(GlobalTracer.get());
```

Comment:
I am concerned with the race conditions here, as you don't know whether TracingUtils class will be loaded before we have a chance to init global tracer. You may need to change this into a lazy dependency, i.e. instead of passing tracer (or active span source, per the other comment) to the TraceContext impl, you mass a _function that returns an active span source_, so that it's dereferenced as needed.

Reply:
I thought this was was fine because `GlobalTracer.get()` returns an instance that is a new reference to itself. However, I agree that we cannot guarantee that TracingUtil is loaded after the tracer is initialized. I'll move this to a lazy init block.